cumulus/client/service/src/lib.rs

```diff
@@ -572,3 +603,133 @@ where
 
 	Err("Stopping following imported blocks. Could not determine parachain target block".into())
 }
+
+async fn parachain_informant<Block: BlockT, Client>(
+	relay_chain_interface: impl RelayChainInterface + Clone,
+	sync_service: Arc<dyn SyncOracle + Send + Sync>,
+	client: Arc<Client>,
+	para_id: ParaId,
+	metrics: Option<ParachainInformantMetrics>,
+) where
+	Client: HeaderBackend<Block> + Send + Sync + 'static,
+{
+	// Backed blocks.
+	let pending_candidates =
+		match pending_candidates(relay_chain_interface.clone(), para_id, sync_service).await {
+			Ok(pending_candidates_stream) => pending_candidates_stream.fuse(),
+			Err(err) => {
+				log::error!("Unable to retrieve pending candidate stream: {err:?}.");
+				return
+			},
+		};
+	let backed_blocks = pending_candidates.flat_map(|(candidates, _, relay_header)| {
+		futures::stream::iter(candidates.into_iter().filter_map(move |candidate| {
+			let relay_header = relay_header.clone();
+			match Block::Header::decode(&mut &candidate.commitments.head_data.0[..]) {
+				Ok(header) => Some((header, relay_header)),
+				Err(e) => {
+					log::warn!("Failed to decode parachain header from backed block: {e:?}");
+					None
+				},
+			}
+		}))
+	});
+	futures::pin_mut!(backed_blocks);
+
+	// Included blocks.
+	let finalized_heads = match finalized_heads(relay_chain_interface.clone(), para_id).await {
+		Ok(finalized_heads_stream) => finalized_heads_stream.fuse(),
+		Err(err) => {
+			log::error!("Unable to retrieve finalized heads stream: {err:?}.");
+			return
+		},
+	};
+	let included_blocks = finalized_heads.filter_map(|(head, relay_header)| async move {
+		match Block::Header::decode(&mut &head[..]) {
+			Ok(header) => Some((header, relay_header)),
+			Err(e) => {
+				log::warn!("Failed to decode parachain header from finalized block: {e:?}");
+				None
+			},
+		}
+	});
+	futures::pin_mut!(included_blocks);
+
+	let mut last_backed_block: Option<<Block as BlockT>::Header> = None;
+	let mut last_backed_block_time: Option<Instant> = None;
+	let mut last_included_block: Option<<Block as BlockT>::Header> = None;
+	loop {
+		select! {
+			(backed_block, relay_block) = backed_blocks.select_next_some() => {
+				if last_backed_block.as_ref() == Some(&backed_block) {
+					// Ignore duplicate notifications.
+					continue;
+				}
+				log::info!(
+					"New backed block at relay block #{} ({}): #{} ({})",
+					relay_block.number(),
+					relay_block.hash(),
+					backed_block.number(),
+					backed_block.hash(),
+				);
+				let backed_block_time = Instant::now();
+				if let Some(last_backed_block_time) = &last_backed_block_time {
+					let duration = backed_block_time.duration_since(*last_backed_block_time);
+					if let Some(metrics) = &metrics {
+						metrics.parachain_block_authorship_duration.observe(duration.as_secs_f64());
+					}
+				}
+				last_backed_block_time = Some(backed_block_time);
+				last_backed_block = Some(backed_block.clone());
```

Comment:
Do we need the clone here?
```suggestion
				last_backed_block = Some(backed_block);
```